File: 프로그래머스/Lv1/3진법 뒤집기.js
// 내 풀이
function solution(n) {
    let answer = "";
    let i = 1;
    let result = 0;

    while(3**i <= n){
        i++;
    }

    for(let j=i-1; j>-1; j--){
        for(let k=2; k>-1; k--){
            if((3**j)*k <= n){
                answer += String(k);
                n -= (3**j)*k;
                break;
            }
        }
    }

    for(let z=0; z<answer.length; z++){
        result += 3**z*answer[z];

    }
    return result;
}

// 다른 사람 풀이
const solution2 = (n) => {
    // {{10진수}}.toString(n) 를 사용해 10진수를 원하는 n진수로 변환
    // parseInt({{n진수}}, n) 를 사용해 n진수를 10진수로 변환
    return parseInt([...n.toString(3)].reverse().join(""), 3);
}

console.log(solution(45));
console.log(solution2(125));

File: 프로그래머스/Lv1/3진법 뒤집기.py
# 내 풀이
def solution(n):
    # n진수  → 10진수 : int(string, base)
    # 10진수  → 2, 8, 16진수 bin(), oct(), hex() 함수를 지원
    answer = ""
    i = 1

    while n >= 3**i:
        i += 1

    for j in range(i-1, -1, -1):
        for k in range(2, -1, -1):
            if 3**j*k <= n : 
                answer += str(k)
                n -= 3**j*k
                break

    answer = answer[::-1]

    return int(answer, 3)

# 다른 사람 풀이
def solution2(n):
    tmp = ''
    while n:
        tmp += str(n % 3)
        n = n // 3

    answer = int(tmp, 3)
    return answer

print(solution(45))
print(solution2(125))